### PR TITLE
Apply free look toggle to current input type only

### PR DESCRIPTION
**Pull request by ceski-1**
Fixes https://github.com/fabiangreffrath/woof/issues/1476

### CODE REVIEW

File: src/m_menu.c [hunk 1]
```diff
@@ -4943,9 +4943,18 @@ static boolean M_ShortcutResponder(void)
...
+            // Gamepad free look toggle only affects gamepad.
+            padlook = !padlook;
+            togglemsg("Gamepad Free Look %s", padlook ? "On" : "Off");
+        }
+        else
+        {
+            // Keyboard or mouse free look toggle only affects mouse.
+            mouselook = !mouselook;
+            togglemsg("Free Look %s", mouselook ? "On" : "Off");
+        }
```
Comment (ceski-1) [inline src/m_menu.c:4957]: This means a keyboard binding for free look only affects the mouse. Seems like a safe assumption.

File: src/m_misc.c [hunk 2]
```diff
@@ -1159,11 +1159,11 @@ default_t defaults[] = {
...
 
   {
-    "input_mouselook",
+    "input_freelook",
     NULL, NULL,
     {0}, {UL,UL}, input, ss_keys, wad_no,
-    "key to toggle mouselook",
-    input_mouselook, { {0, 0} }
+    "key to toggle free look",
+    input_freelook, { {0, 0} }
```
Comment (ceski-1) [inline src/m_misc.c:1166]: Resets the old binding. Too annoying?

File: src/m_menu.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/m_menu.c:4957]: Indeed, seems legit.

File: src/m_misc.c (hunk 2 above)
Comment (fabiangreffrath) [reply src/m_misc.c:1166]: We could just mention it in the changelog, no problem.

**Review by rfomin — APPROVED** [review]

**Pull request merged**